HACKING.md

```diff
@@ -224,10 +224,57 @@ OpenShift is checked into this repository.  To install `godep` locally run:
 
 If you are not updating packages you should not need godep installed.
 
-## Cherry-picking an upstream commit into Origin
+## Cherry-picking an upstream commit into Origin: Why, how, and when.
+
+Upstream origin carries patches inside of vendor/ on top of each rebase.  Thus, Origin consumes Kubernetes in two ways.
+
+1. *periodic rebases* against a Kubernetes commit.  Eventually, any code you have in upstream kubernetes will land in Openshift via this mechanism.
+
+2. Cherry-picked patches for important *bug fixes*.  We really try to limit feature back-porting entirely.
+
+### Manually
+
+You can manually try to cherry pick a commit (by using git apply). This can easily be done in a couple of steps.
+
+- wget the patch, i.e. `wget -O /tmp/mypatch https://github.com/kubernetes/kubernetes/pull/34624.patch`
+- PATCH=/tmp/mypatch git apply --directory vendor/k8s.io/kubernetes $PATCH
+
+If this fails, then its possible you may need to pick multiple commits.
+
+### For Openshift newcomers: Should I cherry pick my kubernetes fix into Openshift ? Or should I wait for the next rebase?
+
+Assuming you read the bullets above... If your patch is really far behind, for example, if there have been 5 commits modifying the directory you care 
+
```

Comment:
Looks like this section got some extra line breaks that are messing up the formatting if you look at the rendered markdown.